core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -260,8 +261,11 @@ private ReactiveProcessor dispatchToFlow() {
       Optional<Object> onErrorStrategy = context.getOrEmpty(KEY_ON_NEXT_ERROR_STRATEGY);
       if (onErrorStrategy.isPresent()
           && onErrorStrategy.get().toString().contains(ON_NEXT_FAILURE_STRATEGY)) {
```

Comment:
I'd check if the strategy corresponds to an error continue by using `errorStrategy.get().getClass().getName()` instead of the `toString()`